smartsim/experiment.py

```diff
@@ -371,6 +371,25 @@ def summary(self, style: str = "github") -> str:
             disable_numparse=True,
         )
 
+    def stop(self, *ids: LaunchedJobID) -> tuple[JobStatus | InvalidJobStatus, ...]:
+        """Cancel the execution of a previously launched job.
+
+        :param ids: The ids of the launched jobs to stop.
+        :raises ValueError: No job ids were provided.
+        :returns: A tuple of job statuses upon cancellation with order
+            respective of the order of the calling arguments.
+        """
+        if not ids:
+            raise ValueError("No job ids provided")
+        by_launcher = self._launch_history.group_by_launcher(set(ids), unknown_ok=True)
+        id_to_stop_stat = (
+            launcher.stop_jobs(*launched).items()
```

Comment:
One thing I am thinking about here is stop_jobs may raise a ``smartsim.error.errors.LauncherJobNotFound``, and if it does, it seems like the process only partially completes.  Should we complete all of the ones that can complete and then (re)raise the error?

Reply:
In theory this cannot happen if the `_launch_history` is never corrupted (e.g. an id is never mapped to a launcher that never issued it), but I do agree that this could become a problem, especially if users start to make their own launchers.

I moved this to a [new issue](https://github.com/CrayLabs/Poseidon/issues/772) where it can be discussed further because I think this may be a problem with other methods (e.g. `get_status`, `wait`) as well as potentially other errors that can arise when stopping over a collection of ids.